### ISSUE TITLE
Composite data source

### ISSUE THREAD

**Issue by jvmlet** [serge-rider/dbeaver#1804]
Hi
Squirrel SQL has **multisource** plugin, is this feature supported by dbeaver ?

**Comment by serge-rider** [comment]
No, there isn't.
What is this plugin about?

**Comment by jvmlet** [comment]
It  allows you to run query against   different databases , like

````
 select a.a from db1.schema1.table1 a
where a.field in ( select b db2.schema2.table2  b)
````
`db1` and `db2` are different **databases**

**Comment by serge-rider** [comment]
As far as I know it is based on Unity JDBC driver (which is a commercial software). It can be used in DBeaver as well as in Squirrel.
Although I didn't try this yet.

**Comment by jvmlet** [comment]
From [here ](http://squirrel-sql.sourceforge.net/index.php?page=plugins):

> Multisource Plugin by Ramon Lawrence and Gerd Wagner
Installer Category: Optional
This plugin allows queries that cross multiple databases such as SELECT * FROM OracleDB.table1 INNER JOIN MySQL.table2.
It also supports cross-database updates and synchronization, data virtualization, and SQL dialect and function translation.

**Comment by serge-rider** [comment]
In fact multisource plugin is a very thin wrapper around UnityJDBC driver. All it does is UI for connections configuration.

From plugin readme:
> Plugin Limits
The plugin source code, like all of SQuirreL, is released under the GNU Lesser General Public License. The UnityJDBC virtualization driver is released under a commercial license. However, the UnityJDBC driver included in the plugin is fully functioning with no time limits allowing an unlimited number of sources and queries. The only limitation is the size of the result set is limited to the first 100 rows. (Note there is no limit on the number of rows extracted from each source. So SELECT COUNT(*) FROM table with a 1 million row table works as it only returns one result row.) Use LIMIT 100 to get the first 100 results of a query.

So, Unity is commercial and it looks like they granted special permission for Squirrel to redistribute some old version of Unity along with multisource plugin. Or something like that.

**Comment by pavelgordon** [comment]
Hi, are there any updates on this feature?

**Comment by sidneydemoraes** [comment]
@pavelgordon I don't see that coming anytime soon for this is a non-commercial project while the necessary driver to perform that only has a commercial version. Until there's an alternative to UnityDriver it would demand a lot of the developer's time.

**Comment by sidneydemoraes** [comment]
Unless someone thinks that can port Squirrel's plugin which is open source and can be found [here](http://unityjdbc.com/squirrel/). In this link you have the open source code along with installation instructions for Squirrel which could help understand how things work.
But again, this will take a lot of time (probably).

**Comment by serge-rider** [comment]
As far as I know Unity driver included in SQuirrel is trial version. So it also not fully functional. 

It is not a big deal to write an plugin similar to SQuirrel one, it is just a simple UI wrapper. But I don't want to do this because UnityJDBC is a commercial proprietary software.
So, making brand new open-source driver seems to be the only possible way...

**Comment by p91paul** [comment]
Since the multisource plugin for SQuirrel is apparently developed by unityjdbc, you could try convincing unityjdbc to write one for dbeaver too. They could then offer it as an eclipse update site and people can download it into dbeaver. And by "you" I mean the unityjdbc users, maybe leveraging your paid license.
Writing an open source equivalent to the whole unityjdbc basically means writing a full fledged DBMS.

**Comment by mordillo123** [comment]
Any updates? Tnx

**Comment by patricksmith-built** [comment]
Any updates would be appreciated, thanks!

**Comment by pkoevesdi** [comment]
I'd like to see this feature as well, and I don't see the problem in general to enable an openSource software like DBeaver to let the users use commercial or trial drivers / libraries.

**Comment by pkoevesdi** [comment]
> As far as I know Unity driver included in SQuirrel is trial version.

I can confirm that:
![grafik](https://github.com/dbeaver/dbeaver/assets/33196971/5c70b1f7-3acb-4d5b-ade5-be957bdc5ca9)
(from Squirrel GUI)

But replacing `unityjdbc.jar` with a trial version from https://unityjdbc.com/ makes it work fully for 30 days. I suppose, replacing it with a registered (paid) version works too, but didn't try - I don't have one. So I'd like to encourage to enable users to use the trial or paid license with DBeaver.

**Comment by pkoevesdi** [comment]
In other words, wouldn't it be nice to have DBeaver in this list:
![grafik](https://github.com/dbeaver/dbeaver/assets/33196971/3092170c-d77b-46b5-9feb-a60c08d4a61c)
(Source: https://unityjdbc.com)

**Comment by ghost** [comment]
is it easy to install  trial version and come back to standart jdbc driver ?
I am really interested by this feature

**Comment by pkoevesdi** [comment]
Yes, as far as I know, it is no "Installing", it is just (re)placing the trial or the open or the commercial .jar file.
